application/src/classes/states/track-loader.js

```diff
@@ -425,29 +425,24 @@ TrackLoaderState.prototype.updateCamera = function()
             averagePlayerPosition[0] += this.cars[i].x;
             averagePlayerPosition[1] += this.cars[i].y;
             carCount += 1;
-        }
 
-        squaredDistance = (
-            Math.pow(this.cars[i].x - nextMarker.x, 2) +
-            Math.pow(this.cars[i].y - nextMarker.y, 2)
-        );
+            squaredDistance = (
+                Math.pow(this.cars[i].x - nextMarker.x, 2) +
+                Math.pow(this.cars[i].y - nextMarker.y, 2)
+            );
 
-        if (squaredDistance < closestSquaredDistance) {
-            closestSquaredDistance = squaredDistance;
-            closestCar             = {
-                x : this.cars[i].x,
-                y : this.cars[i].y
-            };
+            if (squaredDistance < closestSquaredDistance) {
+                closestSquaredDistance = squaredDistance;
+                closestCar             = {
+                    x : this.cars[i].x,
+                    y : this.cars[i].y
+                };
+            }
         }
     }
 
-    if (carCount > 0) {
```

Comment:
This is in here to fix a crash that was caused by a divide-by-zero error when all cars were not visible at once (https://github.com/tdboone/mm-ripoff/commit/896b330823f3d3dd3deaef40ac7468ce9d91e8c4) Maybe instead of taking this out completely you could have the camera just do nothing if there are no cars visible?